src/TracerBudgetTerms.jl

```diff
@@ -0,0 +1,196 @@
+    tracer_index = findfirst(x -> x == tracer_name, keys(model.tracers))
+    @inbounds c = model.tracers[tracer_name]
+    return TracerDiffusion(model, Val(tracer_index), c, model.closure, model.diffusivity_fields, model.clock, fields(model), model.buoyancy; kwargs...)
+end
+
+
+"""
+    $(SIGNATURES)
+
+Calculates the diffusion term due to the bathymetry term as
+
+    DIFF = ∂ⱼ 𝓆ᶜⱼ,
+
+where 𝓆ᶜⱼ is the bathyemtry-led diffusion tensor for tracer `c`, using the Oceananigans' kernel
+`immersed_∇_dot_qᶜ`.
+
+
+```jldoctest
+julia> using Oceananigans, Oceanostics
+
+julia> grid = RectilinearGrid(size=(4, 4, 4), extent=(1, 1, 1));
+
+julia> model = NonhydrostaticModel(; grid, tracers=:a);
+
+julia> DIFF = ImmersedTracerDiffusion(model, :a)
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── kernel_function: immersed_∇_dot_qᶜ (generic function with 2 methods)
+└── arguments: ("4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU", "FluxBoundaryCondition: Nothing", "Nothing", "Nothing", "Val{1}", "Clock{Float64, Float64}(time=0 seconds, iteration=0, last_Δt=Inf days)", "(u=4×4×4 Field{Face, Center, Center} on RectilinearGrid on CPU, v=4×4×4 Field{Center, Face, Center} on RectilinearGrid on CPU, w=4×4×5 Field{Center, Center, Face} on RectilinearGrid on CPU, a=4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU)")
+```
+"""
+function ImmersedTracerDiffusion(model, c, c_immersed_bc, closure, diffusivity_fields, val_tracer_index, clock, model_fields; location = (Center, Center, Center))
+    validate_location(location, "ImmersedTracerDiffusion", (Center, Center, Center))
+    return KernelFunctionOperation{Center, Center, Center}(immersed_∇_dot_qᶜ, model.grid, c, c_immersed_bc, closure, diffusivity_fields, val_tracer_index, clock, model_fields)
+end
+
+function ImmersedTracerDiffusion(model, tracer_name; kwargs...)
+    tracer_index = findfirst(x -> x == tracer_name, keys(model.tracers))
+    tracer = model.tracers[tracer_name]
+    immersed_bc = tracer.boundary_conditions.immersed
+    return ImmersedTracerDiffusion(model, tracer, immersed_bc, model.closure, model.diffusivity_fields, Val(tracer_index), model.clock, fields(model); kwargs...)
+end
+
+@inline total_∇_dot_qᶜ(i, j, k, grid, c, c_immersed_bc, closure, diffusivity_fields, val_tracer_index, clock, model_fields, buoyancy) =
+    ∇_dot_qᶜ(i, j, k, grid, closure, diffusivity_fields, val_tracer_index, c, clock, model_fields, buoyancy) +
+    immersed_∇_dot_qᶜ(i, j, k, grid, c, c_immersed_bc, closure, diffusivity_fields, val_tracer_index, clock, model_fields, buoyancy)
+
+"""
+    $(SIGNATURES)
+
+Calculates the total diffusion term as
+
+    DIFF = ∂ⱼ qᶜⱼ + ∂ⱼ 𝓆ᶜⱼ,
+
+where qᶜⱼ is the interior diffusion tensor and 𝓆ᶜⱼ is the bathyemtry-led diffusion tensor for tracer
+`c`. The calculation is done using the Oceananigans' kernels `∇_dot_qᶜ` and `immersed_∇_dot_qᶜ`.
```

Comment:
[`∇_dot_qᶜ`](https://github.com/CliMA/Oceananigans.jl/blob/2109de4547e3527f6fffa7e40364535fd76fefe2/src/TurbulenceClosures/closure_kernel_operators.jl#L43-L48) and [`immersed_∇_dot_qᶜ`](https://github.com/CliMA/Oceananigans.jl/blob/2109de4547e3527f6fffa7e40364535fd76fefe2/src/TurbulenceClosures/immersed_diffusive_fluxes.jl#L228C1-L229C94)